root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -15,6 +15,106 @@ import {
   Checker,
 } from '../../edit/URLCleanup';
 
+/*
+ * This file tests the cleanups and autoselect / restrictions defined in
+ * root/static/scripts/tests/Control/URLCleanup.js
+ *
+ * The main part of the file is the testData object, which contains
+ * the expected results for all the tests. The following properties
+ * are supported in testData:
+ *
+ * input_url:
+ *      The raw URL "entered by the user" (a string).
+ *
+ *      This is always mandatory.
+ *
+ * input_entity_type:
+ *      The entity type selected by the user (a string, such as 'release').
+ *      This is important because the same URL can be assigned to different
+ *      relationship types (or blocked altogether) depending on the selected
+ *      entity type.
+ *
+ *      Required by 'expected_relationship_type', 'expected_error'
+ *      and 'limited_link_type_combinations'. Forbidden if none are present.
+ *
+ * expected_relationship_type:
+ *      The relationship type (or types) we expect the URL
+ *      to get autoselected to. Either a string, such as 'downloadfree',
+ *      or an array of strings, such as ['downloadfree', 'streamingfree'].
+ *      Can also be set to undefined if autoselection is not supposed
+ *      to happen.
+ *
+ *      Optional. If present, 'input_entity_type' is required.
+ *
+ * limited_link_type_combinations:
+ *      An array of all the possible combinations of relationship types
+ *      allowed for the entity type indicated by 'input_entity_type'. This is
+ *      different from autoselection - some relationship types in this list
+ *      might be autoselected, while others are just available for selection
+ *      to the user. To test for autoselection, also use
+ *      'expected_relationship_type' alongside this.
+ *      Each of the allowed combinations can be just a string
+ *      or an array of strings. For example, a possible value for this
+ *      property would be the following:
+ *      [
+ *        'downloadpurchase',
+ *        'streamingpaid',
+ *        ['downloadpurchase', 'streamingpaid'],
```

Comment:
IIRC the nested array means "both can be selected at the same time" but it'd be nice to mention that explicitly -- I seem to always forget.

Reply:
Done in 13d9e10